packages/block-library/src/image/image.js

```diff
@@ -438,24 +438,39 @@ export default function Image( {
 		[ isSingleSelected ]
 	);
 
+	const hasParentPattern = useSelect(
+		( select ) => {
+			const { getBlockParentsByBlockName } = select( blockEditorStore );
+			return (
+				getBlockParentsByBlockName( clientId, 'core/block' ).length > 0
+			);
+		},
+		[ clientId ]
+	);
```

Comment:
Let's select this value via the existing selector. There's no need to create an extra `block-editor` subscription for every image block.

Reply:
Sure! But just being curious, is adding a single store subscription expensive? If so, then I think we might better fix that at a lower-level and make that fast enough. For instance, `react-redux`'s `useSelector` API doesn't have this problem most of the time.

Reply:
It depends on the use case. It could be expensive for very active stores (block editor) and commonly used blocks (Paragraph/Heading/Image). See https://github.com/WordPress/gutenberg/pull/54819#issuecomment-1761202859.

> For instance, react-redux's useSelector API doesn't have this problem most of the time.

Can you share any resources where I can read more about it?

Reply:
> Can you share any resources where I can read more about it?

It's [recommended in the official doc](https://react-redux.js.org/api/hooks#equality-comparisons-and-updates) to call `useSelector` several times even though that creates multiple subscriptions.

Since selectors should be pure and subscriptions should be batched, they should run pretty fast and should rarely be the performance bottleneck. If we see significant performance issues then it's likely other parts of the code, or we have bugs in our `useSelect` implementation. 🤔